libmamba/include/mamba/core/repo.hpp

```diff
@@ -56,91 +62,52 @@ namespace mamba
     {
     public:
 
-        ~MRepo();
+        MRepo(MPool& pool, const std::string& name, const fs::u8path& filename, const RepoMetadata& meta);
+        MRepo(MPool& pool, const PrefixData& prefix_data);
+        MRepo(MPool& pool, const std::string& name, const std::vector<PackageInfo>& uris);
 
         MRepo(const MRepo&) = delete;
+        MRepo(MRepo&&) = default;
         MRepo& operator=(const MRepo&) = delete;
-
-        MRepo(MRepo&&);
-        MRepo& operator=(MRepo&&);
+        MRepo& operator=(MRepo&&) = default;
 
         void set_installed();
         void set_priority(int priority, int subpriority);
-        void add_package_info(Repodata*, const PackageInfo& pkg_info);
-        void add_pip_as_python_dependency();
-
-        const fs::u8path& index_file();
 
         Id id() const;
-        std::string name() const;
-        bool write() const;
-        const std::string& url() const;
         Repo* repo() const;
-        std::tuple<int, int> priority() const;
-        std::size_t size() const;
-
-        bool clear(bool reuse_ids);
-
-        /**
-         * Static constructor.
-         * @param pool ``libsolv`` pool wrapper
-         * @param name Name of the subdirectory (<channel>/<subdir>)
-         * @param filename Name of the index file
-         * @param url Subdirectory URL
-         */
-        static MRepo&
-        create(MPool& pool, const std::string& name, const std::string& filename, const std::string& url);
-
-        /**
-         * Static constructor.
-         * @param pool ``libsolv`` pool wrapper
-         * @param name Name of the subdirectory (<channel>/<subdir>)
-         * @param index Path to the index file
-         * @param meta Metadata of the repo
-         */
-        static MRepo&
-        create(MPool& pool, const std::string& name, const fs::u8path& filename, const RepoMetadata& meta);
-
-        /**
-         * Static constructor.
-         * @param pool ``libsolv`` pool wrapper
-         * @param prefix_data prefix data
-         */
-        static MRepo& create(MPool& pool, const PrefixData& prefix_data);
-
-        /**
-         * Static constructor.
-         * @param pool ``libsolv`` pool wrapper
-         * @param name Name
-         * @param uris Matchspecs pointing to unique resources (URL or files)
-         */
-        static MRepo&
-        create(MPool& pool, const std::string& name, const std::vector<PackageInfo>& uris);
 
-    private:
+        struct [[deprecated]] PyExtraPkgInfo
```

Comment:
I'm not sure if `[[deprecated]]` is useful here as we'll change the major version?

Reply:
It's mostly for internal use, we'll know to remove it in the major release and it is to indicate that it is not needed internally anymore.